src/main/java/org/jboss/logmanager/formatters/Formatters.java

```diff
@@ -477,6 +483,30 @@ public void renderRaw(Formatter formatter, final StringBuilder builder, final Ex
         };
     }
 
+    private static DateTimeFormatter createDateTimeFormatter(String formatString) {
+        if ((formatString == null) || "ISO8601".equals(formatString)) { // Logback uses 'ISO8601'
+            return DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss,SSS");
+        }
+        final DateTimeFormatter cached = CACHED_DATE_TIME_FORMATTERS.get(formatString);
+        if (cached != null) {
+            return cached;
+        }
+        if (CONSTANT_PATTERN.matcher(formatString).matches()) {
+            DateTimeFormatter result;
+            try {
+                final MethodHandle methodHandle = MethodHandles.publicLookup()
+                        .findStaticGetter(DateTimeFormatter.class, formatString, DateTimeFormatter.class);
```

Comment:
I would advise against using reflection for this. It would tie the supported formats to the JDK in use, which would probably cause an issue at some point if and when we want to support (generally speaking) a format that only exists by name in a later JDK.

Also, the cache is probably overkill given that patterns are seldom created; I think a `String`-switch which resolves a name to the corresponding string or predefined constant is probably best. This allows us to control the namespace and also support formats that do not exist by name in the API, and also provides protection if such a format is introduced under a different name.

Reply:
Okay. That is a valid point. I was trying to think of a generic way to handle it, but it makes sense to keep it hard-coded.